tests/unit/src/test/scala/tests/TypeDefinitionLspSuite.scala

```diff
@@ -0,0 +1,146 @@
+      super.testAsync(name, maxDuration)(run)
+    }
+  }
+
+  def locationFromString(str: String): l.Location = {
+    def positionFromString(str: String): l.Position = {
+      val ints = str.split(":").map(_.toInt)
+      new l.Position(ints.head, ints.last)
+    }
+
+    val split = str.split("""/\\""")
+    val (uri, rest) = (split.head, split.tail)
+    val range =
+      new l.Range(positionFromString(rest.head), positionFromString(rest.last))
+    new l.Location(
+      workspace.toURI.resolve(uri).toString.replaceFirst("/", "///"),
+      range
+    )
+  }
+
+  def createLocation(
+      uri: String,
+      begin: (Int, Int),
+      end: (Int, Int)
+  ): l.Location = {
+    val startPos = new l.Position(begin._1, begin._2)
+    val endPos = new l.Position(end._1, end._2)
+
+    new l.Location(uri, new l.Range(startPos, endPos))
+  }
+
+  testAsync("int") {
+    for {
+      _ <- server.initialize(
+        """/metals.json
+          |{
+          |   "a":{}
+          |}
+        """.stripMargin
+      )
+      _ <- server.assertTypeDefinition(
+        "a/src/main/scala/a/Main.scala",
+        """
+          |object Main {
+          |  val ts@@t: Int = 2
+          |}""".stripMargin,
+        locationFromString(".metals/readonly/scala/Int.scala/\\25:21/\\25:24")
+      )
+    } yield ()
+  }
+
+  testAsync("basic") {
+    cleanWorkspace()
+    for {
+      _ <- server.initialize(
+        """/metals.json
+          |{"a":{}}
+          |
+          |/a/src/main/scala/a/Main.scala
+          |package a
+          |class Main(i: Int) {}
+          |
+          |object Main extends App {
+          |  val test = new Main(1)
+          |}
+        """.stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/Main.scala")
+      _ = server.workspaceDefinitions
+      _ <- server.assertTypeDefinition(
+        "a/src/main/scala/a/Main.scala",
+        """
+          |package a
+          |class Main(i: Int) {}
+          |
+          |object Main extends App {
+          |  val te@@st = new Main(1)
+          |}
+        """.stripMargin,
+        locationFromString("a/src/main/scala/a/Main.scala/\\2:6/\\2:10")
+      )
+    } yield ()
+  }
+
+  testAsync("method", Duration("10 min")) {
+    cleanWorkspace()
+    for {
+      _ <- server.initialize(
+        """/metals.json
+          |{"a":{}}
+          |
+          |/a/src/main/scala/a/Main.scala
+          |package a
+          |class Main(i: Int) {}
+          |
+          |object Main extends App {
+          |  def foo(mn: Main): Unit = {
+          |     println(mn)
+          |  }
+          |}
+        """.stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/Main.scala")
+      _ = server.workspaceDefinitions
+      _ <- server.assertTypeDefinition(
+        "a/src/main/scala/a/Main.scala",
+        """
+          |package a
+          |class Main(i: Int) {}
+          |
+          |object Main extends App {
+          |  def foo(mn: Main): Unit = {
```

Comment:
Let's add a test with cursor on a invocation `foo` - it should go to the method definition.